aws/data_source_aws_dedicated_host.go

```diff
@@ -0,0 +1,104 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwsDedicatedHost() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsAwsDedicatedHostRead,
+
+		Schema: map[string]*schema.Schema{
+
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"instance_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"host_recovery": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"auto_placement": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsAwsDedicatedHostRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+	filters, filtersOk := d.GetOk("filter")
+
+	params := &ec2.DescribeHostsInput{}
+	if filtersOk {
+		params.Filter = buildAwsDataSourceFilters(filters.(*schema.Set))
+	}
+	// var hostIDs []string
+	resp, err := conn.DescribeHosts(params)
+	if err != nil {
+		return err
+	}
+	// If no hosts were returned, return
+	if len(resp.Hosts) == 0 {
+		return fmt.Errorf("Your query returned no results. Please change your search criteria and try again.")
```

Comment:
Use `errors.New` instead of `fmt.Errorf` if you're creating a new error that does not use any format-string identifiers to embed additional data in the string.

Also, error strings should not be capitalized as per https://github.com/golang/go/wiki/CodeReviewComments#error-strings
Weird rule but whatever. Gopher do what Gopher want.